client/src/main/java/com/tigrisdata/db/client/AbstractTigrisCollection.java

```diff
@@ -92,6 +97,20 @@ protected Iterator<T> readInternal(
     }
   }
 
+  protected Iterator<SearchResult<T>> searchInternal(
+      SearchRequest request, SearchRequestOptions options) throws TigrisException {
+    Api.SearchRequest apiSearchRequest =
+        TypeConverter.toSearchRequest(databaseName, collectionName, request, objectMapper);
+    try {
+      Iterator<Api.SearchResponse> resp = blockingStub.search(apiSearchRequest);
+      Function<SearchResponse, SearchResult<T>> converter =
+          r -> SearchResult.from(r, objectMapper, collectionTypeClass);
+      return Utilities.transformIterator(resp, converter);
+    } catch (StatusRuntimeException ex) {
+      throw new TigrisException(SEARCH_FAILED, ex);
```

Comment:
as discussed please take care of passing error detail to user here in this PR.

Reply:
Will address in next iteration